src/index.js

```diff
@@ -155,28 +156,32 @@ const compile = (schema, root, reporter, opts, scope, basePathRoot) => {
   fun.write('function validate(data) {')
   // Since undefined is not a valid JSON value, we coerce to null and other checks will catch this
   fun.write('if (data === undefined) data = null')
-  if (reporter === true) fun.write('validate.errors = null')
+  if (optIncludeErrors) fun.write('validate.errors = null')
   fun.write('let errors = 0')
 
   const basePathStack = basePathRoot ? [basePathRoot] : []
-  const visit = (allErrors, reporter, name, node, schemaPath) => {
-    const rule = (...args) => visit(allErrors, reporter, ...args)
+  const visit = (allErrors, includeErrors, name, node, schemaPath) => {
+    const rule = (...args) => visit(allErrors, includeErrors, ...args)
     const subrule = (...args) => visit(true, false, ...args)
+    const writeError = (format, ...params) => {
+      if (allErrors) {
+        fun.write('if (validate.errors === null) validate.errors = []')
+        fun.write(`validate.errors.push(${format})`, ...params)
+      } else {
+        fun.write(`validate.errors = [${format}]`, ...params)
+        fun.write('return false')
+      }
+    }
```

Comment:
```suggestion
    const writeError = (format, ...params) => {
      fun.write('if (validate.errors === null) validate.errors = []')
      fun.write(`validate.errors.push(${format})`, ...params)
      if (!allErrors) {
        fun.write('return false')
      }
    }
```

Reply:
That is deliberate.

The form used here generates smaller code and is _significantly_ faster in bechmark for whatever reason (if errors are enabled).

Reply:
I added a comment for clarity.